online-shop/src/app/components/edit-product/edit-product.component.html

```diff
@@ -0,0 +1,29 @@
+<form [formGroup]="formData" (ngSubmit)="submit()">
+  <div>
+    <h1>Edit Product</h1>
+    <button type="submit" [disabled]="!formData.valid">Save</button>
+    <button (click)="cancel()">Cancel</button>
+  </div>
+
+  <table class="clearfix">
+    <tr>
+      <th>Name:</th>
+      <input type="text" name="name" formControlName="name" required minlength="3" />
```

Comment:
I've set it here aswell because I've read the following in a tutorial: 

> Notice that the required attribute is still present in the template. Although it's not necessary for validation, it should be retained to for accessibility purposes.
[https://angular.io/guide/form-validation](url)  

How should I proceed?

Reply:
The **required** attribute is redundant in the HTML file, the validator **Validators.required** handles the accessibility purposes as well, the only difference is that you add the validator from the typescript file such that you are able to do extra checks on a FormControl object.

Reply:
Thank you for clarifying. I have removed them from the HTML files.